pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -132,8 +132,12 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 
 		// Update the status.conditions. This only works the first time, so if you fix an issue, it will still be set to PlatformXXXMisConfigured
 		setStatusCondition(&hyd, hypdeployment.PlatformConfigured, metav1.ConditionFalse, "Configuring platform with infra-id: "+hyd.Spec.InfraID, hypdeployment.BeingConfiguredReason)
-		if hyd.Spec.Infrastructure.Platform != nil && hyd.Spec.Infrastructure.Platform.AWS != nil {
-			setStatusCondition(&hyd, hypdeployment.PlatformIAMConfigured, metav1.ConditionFalse, "Configuring platform IAM with infra-id: "+hyd.Spec.InfraID, hypdeployment.BeingConfiguredReason)
+		if hyd.Spec.Infrastructure.Platform != nil {
+			if hyd.Spec.Infrastructure.Platform.AWS != nil {
+				setStatusCondition(&hyd, hypdeployment.PlatformIAMConfigured, metav1.ConditionFalse, "Configuring platform IAM with infra-id: "+hyd.Spec.InfraID, hypdeployment.BeingConfiguredReason)
+			} else if hyd.Spec.Infrastructure.Platform.Azure != nil {
+				setStatusCondition(&hyd, hypdeployment.PlatformIAMConfigured, metav1.ConditionTrue, "Configuring platform IAM with infra-id: "+hyd.Spec.InfraID, hypdeployment.NotApplicableReason)
```

Comment:
We should talk more about this, as it shows `NA`, but the `READY` = True in the print column output which is weird.